presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AggregationNode.java

```diff
@@ -78,8 +80,13 @@ public AggregationNode(
         this.step = requireNonNull(step, "step is null");
         this.hashSymbol = requireNonNull(hashSymbol, "hashSymbol is null");
         this.groupIdSymbol = requireNonNull(groupIdSymbol, "groupIdSymbol is null");
+        this.rowTypeSymbol = requireNonNull(rowTypeSymbol, "rowTypeSymbol is null");
+        checkArgument(step.isOutputPartial() || !rowTypeSymbol.isPresent(), "rowTypeSymbol is required with partial output");
```

Comment:
is this check correct? It looks like it will pass if step is partial AND rowTypeSymbol is missing.